src/Shared/StringSplit/StringSplitExtensions.cs

```diff
@@ -0,0 +1,891 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if !NET8_0_OR_GREATER
+
+using System;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Shared.StringSplit;
+
+#if !SHARED_PROJECT
+[System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+#endif
+internal static class StringSplitExtensions
+{
+    /// <summary>
+    /// Splits a string into a number of string segments.
+    /// </summary>
+    /// <param name="input">The string to split.</param>
+    /// <param name="separator">A character that delimits the substrings in this instance.</param>
+    /// <param name="result">A span to receive the individual string segments.</param>
+    /// <param name="numSegments">The number of string segments copied to the output.</param>
+    /// <param name="options">A bitwise combination of the enumeration values that specifies whether to trim substrings and include empty substrings.</param>
+    /// <returns><see langword="true" /> if there was enough space in the output array, otherwise <see langword="false" />.</returns>
```

Comment:
```suggestion
    /// <returns><see langword="true" /> if there was enough space in the output array; otherwise, <see langword="false" />.</returns>
```

The same below.

Reply:
Done